### DISCUSSION TITLE
Possible to pass name prop from Controller to render method?

### DISCUSSION THREAD

**Discussion by lsbyerley** [react-hook-form/react-hook-form#2560] [Ideas]
When using the render prop for the Controller component, the name attribute in the Controller component does not get passed down to the component that is rendered by the render function. The example shows an example. The input does not have the name attribute at all. This can be a problemn when using a config file to render your form components and the render function is part of the config file.

```
<Controller
  name={`test[${index}].lastName`}
  control={control}
  defaultValue={item.lastName} // make sure to set up defaultValue
  render={({ onChange, value, label }) => <input onChange={onChange} value={value} />}
/>
```

codesandbox with the issue: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-xn8lx?file=/src/index.js:1099-1392

**Comment by bluebill1049** [comment]
is you want `name` to pass down to your input, you can use `as`. as right now, render doesn't pass down `name`.

**Comment by lsbyerley** [comment]
Yes I realize it does not pass down name to the render function which I why posted this in the "Ideas" section.

It is not possible for me to use `as` in the scenario I'm working with. I must use the render function

**Reply by bluebill1049** [reply]
oh ok, no worries. I am creating a PR to pass-through name.

**Reply by bluebill1049** [reply]
would you like to send a feature request on this one?

**Reply by lsbyerley** [reply]
By send feature request do you mean submit a PR?

**Reply by bluebill1049** [reply]
as sending an issue (feature request) :) my PR can close that ticket as a reference.

**Reply by glebdv** [reply]
@bluebill1049 was there an update for this? We made a switch to react-hook-form recently, and it broke all of our tests that relied on "name" (id also doesnt seem to be passed down). We can rewrite our tests, but it's a bit long ^^ we're running v 7.36.0

**Reply by bluebill1049** [reply]
pretty sure name is returned by the render function. please refer to the doc: https://react-hook-form.com/api/usecontroller
